### ISSUE TITLE
UselessVariable: false positives with returns and assigments in ifs

### ISSUE THREAD

**Issue by carusogabriel** [slevomat/coding-standard#423]
```php
        if ($foo = $this->bar($quo)) {
            $this->baz = true;

            return $foo;
        }
```

Get "fixed" into:

```php
        if (return $this->bar($quo)) {
            $this->baz = true;
        }
```

**Comment by kukulich** [comment]
It should be already fixed: 
https://github.com/slevomat/coding-standard/issues/419

**Comment by carusogabriel** [comment]
Gonna test the dev version and close if it's fixed :smile:

**Comment by kukulich** [comment]
@carusogabriel I'm not able to reproduce it.

**Comment by carusogabriel** [comment]
It was fixed with #419 indeed :)

**Comment by carusogabriel** [comment]
Looks like it wasn't. Code to reproduce:
```php
        if ($foo = $this->foo->bar($baz['_id'], $quoz)) {
            $this->bazo->lotus('Message');

            return $foo;
        }
```

**Comment by carusogabriel** [comment]
And invalidate my comment https://github.com/slevomat/coding-standard/issues/423#issuecomment-416395006, it wasn't fixed.

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/2654744cbcb7c25dbaafce3bc6a91f87c71c27a4
